usd/src/usd_parser/USDJoints.cc

```diff
@@ -0,0 +1,276 @@
+          _prim.GetAttribute(
+            pxr::TfToken("drive:linear:physics:damping")).Get(&damping);
+          _prim.GetAttribute(
+            pxr::TfToken("drive:linear:physics:maxForce")).Get(&maxForce);
+        }
+        else
+        {
+          _prim.GetAttribute(
+            pxr::TfToken("drive:angular:physics:stiffness")).Get(&stiffness);
+          _prim.GetAttribute(
+            pxr::TfToken("drive:angular:physics:damping")).Get(&damping);
+          _prim.GetAttribute(
+            pxr::TfToken("drive:angular:physics:maxForce")).Get(&maxForce);
+        }
+        _prim.GetAttribute(
+          pxr::TfToken("physxJoint:maxJointVelocity")).Get(&vel);
+
+        pxr::UsdAttribute jointFrictionAttribute;
+        if (jointFrictionAttribute = _prim.GetAttribute(
+          pxr::TfToken("physxJoint:jointFriction")))
+        {
+          jointFrictionAttribute.Get(&jointFriction);
+        }
+        else if (jointFrictionAttribute = _prim.GetAttribute(
+          pxr::TfToken("jointFriction")))
+        {
+          jointFrictionAttribute.Get(&jointFriction);
+        }
+
+        jointAxis.SetDamping(damping);
+        jointAxis.SetEffort(maxForce);
+        jointAxis.SetSpringStiffness(stiffness);
+        jointAxis.SetFriction(jointFriction);
+        jointAxis.SetMaxVelocity(vel);
+      }
+
+      if (_prim.IsA<pxr::UsdPhysicsFixedJoint>())
+      {
+        auto variant_physics_fixed_joint = pxr::UsdPhysicsFixedJoint(_prim);
+
+        joint.SetType(sdf::JointType::FIXED);
+
+        return joint;
+      }
+      else if (_prim.IsA<pxr::UsdPhysicsPrismaticJoint>())
+      {
+        auto variant_physics_prismatic_joint =
+          pxr::UsdPhysicsPrismaticJoint(_prim);
+
+        joint.SetType(sdf::JointType::PRISMATIC);
+
+        auto errors = jointAxis.SetXyz(-(q2 * axisVector).Round());
+        if (!errors.empty())
+        {
+          std::cerr << "Errors encountered when setting xyz of prismatic "
+                    << "joint axis:\n";
+          for (const auto &e : errors)
+            std::cerr << e << "\n";
+        }
+
+        joint.SetRawPose(
+          ignition::math::Pose3d(
+            ignition::math::Vector3d(trans[0], trans[1], trans[2]),
+            ignition::math::Quaterniond(q1 * q2)));
+
+        jointAxis.SetLower(lowerLimit * metersPerUnit);
+        jointAxis.SetUpper(upperLimit * metersPerUnit);
+        joint.SetAxis(0, jointAxis);
+
+        return joint;
+      }
+      else if (_prim.IsA<pxr::UsdPhysicsRevoluteJoint>())
+      {
+        auto variant_physics_revolute_joint = pxr::UsdPhysicsRevoluteJoint(_prim);
+
+        joint.SetType(sdf::JointType::REVOLUTE);
+
+        auto errors = jointAxis.SetXyz(axisVector);
+        if (!errors.empty())
+        {
+          std::cerr << "Errors encountered when setting xyz of revolute "
+                    << "joint axis:\n";
+          for (const auto &e : errors)
+            std::cerr << e << "\n";
```

Comment:
Same comments apply here about creating a `UsdError` object for this instead of printing errors to the console